README.md

```diff
@@ -1,35 +1,103 @@
+| TENANT_ID     | <TENANT_ID provided by Würth Phoenix>           |
+| SERVICE_URL   | https://noitest.crm4.dynamics.com/api/data/v9.2 |
+| SERVICE_SCOPE | https://noitest.crm4.dynamics.com/.default      |
+
+***Question**: The SERVICE_URL and SERVICE_SCOPE (...noitest...) will change in the future?*
+
+-----
+
+The following AWS command [configures the environment variables](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-api) for an AWS Lambda function:
+
+```sh
+aws lambda update-function-configuration --function-name my-function \
+    --environment "Variables={BUCKET=my-bucket,KEY=file.txt}"
+```
 
 ## Development
 
-You can develop the app with your coding editor of your choice that supports C#. For example to code with VS Code:
+The project can be developed purely with a basic editor in combination with the dotnet CLI.
+
+### From the command line
+
+Notable commands are:
+
+- `dotnet tool restore`:
+  restores the CLI tools
+- `dotnet restore`:
+  restores the project dependencies
+- `dotnet build`:
+  run a build
+- `dotnet test`:
+  run the tests
+- `dotnet watch test -p ./test/it.bz.noi.community-api.Tests/`:
+  watch and run tests after a file gets modified
+
+### With an IDE
+
+You can develop the app with your coding editor of your choice that supports C#. For example to code with Visual Studio Code:
 
-* Start VS Code, e.g. by launching `code .` from the command line.
-* Install the `ms-dotnettools.csharp` VS Code extension for C# coding support.
+- Start VS Code, e.g. by launching `code .` from the command line.
+- Install the `ms-dotnettools.csharp` VS Code extension for C# coding support (if not already installed).
+- The editor tooling should now work and you can start to code.
 
-## Local testing
+### Local testing
 
 As the AWS documentation states the best way to test the functionality of the app is by writing unit tests then run `dotnet test` (https://aws.amazon.com/de/blogs/developer/net-5-aws-lambda-support-with-container-images/).
 
 I am not convinced that this is a time efficient way to develop and test the functionality so I will try to get it to work via docker-compose and an iterative development approach. Documentation will follow.
 
-## Deploy
-
-To deploy the application the best way is to configure a CI system that execute a deployment on every push to a special `release` tag.
+## Deployment
 
 To deploy the image from your own machine make sure to configure an AWS named profile (https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html).
-Then run the following command:
 
+Then run the following commands:
+
+```sh
+dotnet tool restore
 ```
+
+This restores the lambda command line tool needed for deployment.
+
+Then run the lambda serverless deployment tool:
+
+```sh
 dotnet lambda deploy-serverless it-bz-noi-community--api -sb it-bz-noi-community--api -pl ./src/it.bz.noi.community-api/
 ```
```

Comment:
For me just this command is not clear. What do the two paramters `it-bz-noi-community--api` mean? Where and for what will they be used?

This is important for me, because I for example have to make a test and a production environment and so I cannot name both the same way.

Reply:
In addition, the command above for configuring the environment variables uses the function name `my-function`. Does this function name not have to be related to one of these names? If yes, then I would just the readme accordingly and second describe it more precicely.

Reply:
`it-bz-noi-community--api` is, as you guessed correctly the server function. This is how it will be named on AWS Lambda.
> Does this function name not have to be related to one of these names?
Exactly, it's always the same name. Could you also check if you can make it work with the single command as I described since there seems to be a discrepancy between what I found out about the command and Tobias did.

Reply:
I asked these questions also to show, that I think it would be good that the readme shows/answers these points. I think this should be documented or not?

Reply:
Correct, the `my-function` has to be replaced with the Lambda function instance name.
I intentionally didn't use that name, as it may vary. For my case the name of the function got auto generated.
But on a production deployment the name should definitely be permanent.

Some devops magic is needed here!

![](https://media.giphy.com/media/L3FOOWlFjJkdCP1JPg/giphy.gif)